src/hooks/__tests__/useFormatNumber.test.ts

```diff
@@ -0,0 +1,20 @@
+import formatNumber from "../../hooks/useFormatNumber";
+const enDash = "\u2013";
+
+describe("useFormatNumber hook", () => {
+  it("returns a number with commas and handles a range of numbers", () => {
+    expect(formatNumber("4382")).toEqual("4,382");
+    expect(formatNumber("4382", "1234")).toEqual(`1,234${enDash}4,382`);
+    expect(formatNumber("4382", "XX1234")).toEqual(null);
+    expect(formatNumber("4382XXX", "1234")).toEqual(null);
+    expect(formatNumber("4382XXX", "1234XXX")).toEqual(null);
```

Comment:
Break these expectations out into their own test case and also verify that the warning is being logged.